oozie-to-airflow/mappers/spark_mapper.py

```
@@ -45,190 +59,171 @@ def __init__(
+            name_node = prop.find(SPARK_TAG_NAME)
+            value_node = prop.find(SPARK_TAG_VALUE)
             if name_node is not None and name_node.text and value_node is not None and value_node.text:
                 conf_dict[name_node.text] = value_node.text
         return conf_dict
 
-    def update_class_spark_opts(self, spark_opts_node: ET.Element):
+    @staticmethod
+    def _parse_spark_opts(spark_opts_node: ET.Element):
         """
         Some examples of the spark-opts element:
-
-        '--conf key=value'
-        '--conf key1=value1 value2'
-        '--conf key1="value1 value2"'
-        '--conf key1=value1 key2="value2 value3"'
-        '--conf key=value --verbose --properties-file user.properties'
+        --conf key1=value
+        --conf key2="value1 value2"
         """
+        conf = {}
         if spark_opts_node.text:
             spark_opts = spark_opts_node.text.split("--")[1:]
         else:
-            raise Exception("Spark opts node has no text: {}".format(spark_opts_node))
+            raise ParseException("Spark opts node has no text: {}".format(spark_opts_node))
         clean_opts = [opt.strip() for opt in spark_opts]
         clean_opts_split = [opt.split(maxsplit=1) for opt in clean_opts]
 
-        if ["verbose"] in clean_opts_split:
-            self.__dict__["verbose"] = True
-            clean_opts_split.remove(["verbose"])
-
         for spark_opt in clean_opts_split:
             # Can have multiple "--conf" in spark_opts
             if spark_opt[0] == "conf":
                 # Splits key1=value1 into [key1, value1]
-                conf_val = spark_opt[1].split("=", maxsplit=1)
-                self.conf[conf_val[0]] = conf_val[1]
-            else:
-                self.__dict__[spark_opt[0]] = "'" + " ".join(spark_opt[1:]) + "'"
+                conf_pairs = spark_opt[1].split("=", 1)
+                # Value is required
+                if len(conf_pairs) != 2:
+                    continue
```

Should we fail here? I think this is one of the semantic checks that we should not silently ignore.

---

I updated a PR. Thanks.